src/helpers.ts

```diff
@@ -56,29 +56,32 @@ function hasNodeExecutable(rootPath: string, executable: string): boolean {
 }
 
 /**
- *  Handles getting the jest runner, handling the OS and project specific work too
+ * Handles getting the jest runner, handling the OS and project specific work too
  *
  * @returns {string}
  */
-export function pathToJest(pluginSettings: IPluginSettings) {
-  if (pluginSettings.pathToJest) {
-    if (isBootstrappedWithCRA(pluginSettings.rootPath)) {
-      return 'npm test --'
-    }
-    return normalize(pluginSettings.pathToJest)
+export function pathToJest({ pathToJest, rootPath }: IPluginSettings) {
+  if (isNotDefaultPathToJest(pathToJest)) {
+    return normalize(pathToJest)
   }
 
-  const platform = process.platform
-  if (platform === 'win32' && existsSync(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest.cmd'))) {
-    return normalize(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest.cmd'))
-  } else if (
-    (platform === 'linux' || platform === 'darwin') &&
-    existsSync(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest'))
-  ) {
-    return normalize(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest'))
+  if (isBootstrappedWithCreateReactApp(rootPath)) {
+    return 'npm test --'
```

Comment:
Shouldn't we here also add the `.cmd` file extension on Windows? (at least until #297 hopefully solves all extension related issues)

Reply:
Yes, this is a known problem but that's not what the PR was for. It was intended to be a fast fix. It's the same behaviour as Line 65 on the left, and the same before we've introduced the regression ([here](https://github.com/jest-community/vscode-jest/blob/7964a9d6c675502f8e3f080a3335c975a0908e63/src/helpers.ts#L69)).

Reply:
I bumped into this problem today while testing another PR. I agree #324 was flawed, but to override `jest.pathToJest` without checking if it has already been customized by users will be problematic too, for example, consider for yarn users whose jest.pathToJest was set to `yarn test`, now it will be **silently** override...

If our goal is to provide a "default" setting when `jest.pathToJest` is null, then let's do just that, no more and no less. In general, we should never override users' custom config programmatically (we can suggest, but they should be the one to carry out the change, simply because we could be wrong) and definitely not silently. Actually, giving `jest.pathToJest` is so critical, if we did end up using the default value, I think we should make this information available some where (maybe a new channel: vscode-jest?), we could also use the new channel to communicate the plugin specific info such as workspace info, settings, recommedations etc. 

BTW, this PR has been sitting for a while, what is the plan? People are adopting custom `jest.pathToConfig` that made the regression issue less urgent, but nevertheless we could take this opportunity to improve quality and reduce confusion...

Reply:
> I could pick up your change and test them together (except windows) before we cut the next release…

I would like to also include a PR like #297, such that all pathToJest and Windows problems should be fixed. But that should be a matter of minutes as soon as we can merge this PR.

Reply:
Let's get them all in, I'll handle the jest ones and get that released in a few days - can we try get those two merged in that timeframe and we'll ship a new release?

Reply:
Sure. Can we merge this one or is there something speaking against?

Reply:
@stephtr this PR has no `jest-editor-supports` dependency, if you and @seanpoulter all have tested it and are satisfied with it, then it's probably safe to merge, IMHO.

@orta If you can merge [facebook/jest/6586](https://github.com/facebook/jest/pull/6586) and cut a beta jest release, then we can proceed on this end for the PRs depending on the new `jest-editor-supports`.

Reply:
OK, I've requested a release for editor-support - so let's get this PR in then, and https://github.com/jest-community/vscode-jest/pull/297 can happen whenever